discord/state.py

```diff
@@ -875,12 +875,15 @@ def parse_thread_update(self, data: gw.ThreadUpdateEvent) -> None:
             return
 
         thread_id = int(data['id'])
+        raw = RawThreadUpdateEvent(data)
         thread = guild.get_thread(thread_id)
+        raw.thread = thread
         if thread is not None:
             old = copy.copy(thread)
             thread._update(data)
             if thread.archived:
                 guild._remove_thread(thread)
+            self.dispatch('raw_thread_update', raw)
```

Comment:
This dispatch call should be outside of the `if` block as otherwise the event is only dispatched when the thread is cached.